In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv('../../Data/raws/WA_Fn-UseC_-Telco-Customer-Churn.csv')

In [11]:
print(df['Partner'].unique()) 
print("\n")
print(df['Dependents'].unique())  

['Yes' 'No']


['No' 'Yes']


In [19]:
((df['Dependents'] == 'Yes') & (df['Partner'] == 'No')).sum()

np.int64(361)

In [20]:
(
    (df['Dependents'] == 'Yes') &
    (df['Partner'] == 'No')
).sum() / (df['Dependents'] == 'Yes').sum()

np.float64(0.17109004739336492)

ต้องระวัง! Partner กับ Dependent มีความไม่สัมพันทธ์กัน 361 อาจจะ เป็นกลุ่มคนที่ (พ่อ/แม่เลี้ยงเดี่ยว, คนดูแลพ่อแม่, หย่าร้างแต่ยังมีลูก, อื่น ๆ)